packages/flutter/lib/src/material/switch.dart

```diff
@@ -322,6 +326,41 @@ class Switch extends StatelessWidget {
   /// | Disabled | `Colors.black12`                | `Colors.white10`                |
   final MaterialStateProperty<Color?>? trackColor;
 
+  /// {@template flutter.material.switch.thumbImage}
+  /// The image to use on the thumb of this switch
+  ///
+  /// Resolved in the following states:
+  ///  * [MaterialState.selected].
+  ///  * [MaterialState.hovered].
+  ///  * [MaterialState.focused].
+  ///  * [MaterialState.disabled].
+  ///
+  /// {@tool snippet}
+  /// This example resolves the [thumbImage] based on the current
+  /// [MaterialState] of the [Switch], providing a different [ImageProvider] when it is
+  /// [MaterialState.disabled].
+  ///
+  /// ```dart
+  /// Switch(
+  ///   value: true,
+  ///   onChanged: (_) => true,
+  ///   thumbImage: MaterialStateProperty.resolveWith<ImageProvider>((Set<MaterialState> states) {
+  ///     if (states.contains(MaterialState.disabled)) {
+  ///       return MemoryImage(Uint8List.fromList(<int>[1, 2]));
```

Comment:
For this example, perhaps just have `disabledImage` and 'normalImage` variables for the images so that we don't have to complicate the example with the details of MemoryImage.

Reply:
Small question: do you really want `thumbImage`? Can't I use `Icon(Icons.check)` to do the check?